packages/xstate-react/src/useSpawn.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from 'react';
+import { ActorRef, Behavior, EventObject } from 'xstate';
+import { spawnBehavior } from 'xstate/src/behavior';
+
+/**
+ * React hook that spawns an `ActorRef` with the specified `behavior`.
+ * The returned `ActorRef` can be used with the `useActor(actorRef)` hook.
+ *
+ * @param behavior The actor behavior to spawn
+ * @returns An ActorRef with the specified `behavior`
+ */
+export function useSpawn<TState, TEvent extends EventObject>(
+  behavior: Behavior<TEvent, TState>
+): ActorRef<TEvent, TState> {
+  const actorRef = useMemo(() => {
```

Comment:
This should `useConstant` instead of `useMemo` because the created `actorRef` is stateful. Unfortunately, `useMemo` is an optimization "hint" and not a semantic guarantee - and we don't want to lose state when fast refreshing and stuff.

`useConstant` also has its cons - the stored value won't react to input changes (`behavior` changes in this code) but it's not a must for us to support this (and also there is a high chance that we even shouldn't support it). Given cons of both solutions I would definitely should `useConstant` over `useMemo` here

Reply:
Do we want the actor to be restarted (with a new actor) when the `behavior` changes?

Reply:
I don't think so - we currently do not restart with a new machine if the config changes between `useMachine`/`useInterpret` re-renders.

Reply:
2628fad6344437376d219489ea55278af76f2ce7

Reply:
2628fad6344437376d219489ea55278af76f2ce7